README.md

```diff
@@ -1,15 +1,203 @@
-- Compact and simple code base. Reduce the barrier of understanding of source code for easier contributing.
-- Good interop with other crates. `tokio`(for async runtime) and `http`(for http types) are used directly as dependency.
+<details>
+  <summary><b>Table of Contents</b></summary>
+  <p>
 
-[^1]: `http-file` excluded due to design issue.
-[^2]: only guaranteed inside project's own code. unsafe can and is used by dependencies
+  - [🚀 Features](#features-)
+  - [📊 Minimum Supported Rust Version](#minimum-supported-rust-version-)
+  - [⚡ Quick Start](#quick-start-)
+  - [🗨️ FAQ (Frequently Asked Questions)](#faq-frequently-asked-questions-%EF%B8%8F)
+  - [📣 More Contributors Wanted](#more-contributors-wanted-)
+  - [💖 Supporting Xitca-web](#supporting-xitca-web-)
+  - [📘 Documentation](#documentation-)
+  - [🛣️ Roadmap](#roadmap-%EF%B8%8F)
+  - [🙋 Contributing](#contributing-)
+  - [📜 License](#license-)
+  - [🤝 Credits](#credits-)
+
+  </p>
+</details>
+
+# Features 🚀
+
+- 🎗️ Supports HTTP/1.x and HTTP/2
+- 💪 Powerful request routing with optional opt-in macros
+- 📼 Full Tokio compatibility
+- 📦 Cross crate integration with Tower.
+- 🔌 Client/server WebSockets support
+- 🗜️ Transparent content compression/decompression (br, gzip, deflate, zstd)
+- 🚰 Multipart streams
+- 📄 Static assets
+- 🔒 SSL support using Xitca-tls or rustls
+- 🔌 Middlewares (Logger, Tracing, etc)
+
+**[⬆️ Back to Top](#xitca-web)**
+
+# Minimum Supported Rust Version 📊
+
+The latest release of the crate supports 1.79 and above rust versions.
+
+**[⬆️ Back to Top](#xitca-web)**
+
+# Quick Start ⚡
+
+> For a list of other examples, see: [**Examples**](examples)
+
+To get started using Xitca-web add the following code to your `main.rs` file: 
+```rust
+use xitca_web::{handler::handler_service, middleware::Logger, route::get, App};
+
+async fn index() -> &'static str {
+    "Hello world!!"
+}
+
+#[tokio::main]
+async fn main() -> std::io::Result<()> {
+    App::new()
+        .at("/", get(handler_service(index)))
+        .enclosed(Logger::new())
+        .serve()
+        .bind("localhost:8080")?
+        .run()
+        .await
+}
+```
+
+And then add the following dependencies to your `Cargo.toml` file:
+
+```toml
+xitca-web = { version = "*" , features = ["logger"]}
+tokio = { version = "*", default-features = false, features = ["full"] }
+```
+
+And then run the application/server using `cargo run` command and it should start the server/application on <http://127.0.0.1:8080>. Open your prefered browser and navigate to the address on which the server has started and you should see a `hello world!!` message appear.
+
+> [!Warning]
+> The project is still not stable. So expect breaking changes to appear at any time.
+
+**[⬆️ Back to Top](#xitca-web)**
+
+# FAQ (Frequently Asked Questions) 🗨️
+
+## Why Yet Another HTTP Framework?
```

Comment:
It will be better if this part explain the design motivation of xitca-web.(the points in previous readme)
It should also mention that `xitca-web` as a project is not only a web framework. It includes it's own http library, async IO abstraction, tls integration and even a database driver.

Reply:
> It will be better if this part explain the design motivation of xitca-web.(the points in previous readme) It should also mention that `xitca-web` as a project is not only a web framework. It includes it's own http library, async IO abstraction, tls integration and even a database driver.

Ok I see, I didn't quite knew about them but I will mention these in this section. :slightly_smiling_face: